### PR TITLE
For #16807 - Make sure there's only one instance of search dialog

### PR DESCRIPTION
**Pull request by ekager**
### Pull Request checklist

- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features. In addition, it includes a screenshot of a successful [accessibility scan](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor&hl=en_US) to ensure no new defects are added to the product.

### To download an APK when reviewing a PR:
1. click on Show All Checks,
2. click Details next to "Taskcluster (pull_request)" after it appears and then finishes with a green checkmark,
3. click on the "Fenix - assemble" task, then click "Run Artifacts".
4. the APK links should be on the left side of the screen, named for each CPU architecture

### CODE REVIEW

**Comment by gabrielluong** [comment]
I tested this with the following STR:
1. Open a webpage
2. Open the tab trays
3. Go to Android Home
4. Click on the search widget

I still get the described actual behaviour in https://github.com/mozilla-mobile/fenix/issues/16807 where the search fragment isn't actually shown and the awesomebar isn't functional until I do a "back" to navigate back in the back stack.

File: app/src/main/res/navigation/nav_graph.xml [hunk 1]
```diff
@@ -25,7 +25,9 @@
 
     <action
         android:id="@+id/action_global_search_dialog"
-        app:destination="@id/searchDialogFragment" />
+        app:destination="@id/searchDialogFragment"
```
Comment (gabrielluong) [inline app/src/main/res/navigation/nav_graph.xml:28]: I tested this with the following STR:
1. Open a webpage
2. Open the tab trays
3. Go to Android Home
4. Click on the search widget

I still get the described actual behaviour in https://github.com/mozilla-mobile/fenix/issues/16807 where the search fragment isn't actually shown and the awesomebar isn't functional until I do a "back" to navigate back in the back stack.

**Review by gabrielluong — CHANGES_REQUESTED** [review]

File: app/src/main/res/navigation/nav_graph.xml (hunk 1 above)
Comment (ekager) [reply app/src/main/res/navigation/nav_graph.xml:28]: lol I totally misunderstood the original bug and was just fixing another apparently where you can get 2 search dialogs overlying eachother.

**Comment by ekager** [comment]
The bug I fixed STR was 
1. Click URL to open search dialog
2. Swipe away to Android home
3. Click search widget

This updated PR should fix both of these bugs

File: app/src/main/res/navigation/nav_graph.xml [hunk 2]
```diff
@@ -25,7 +25,9 @@
...
+        app:popUpTo="@id/searchDialogFragment"
+        app:popUpToInclusive="true"/>
```
Comment (eliserichards) [inline app/src/main/res/navigation/nav_graph.xml:30]: Perfect! Good catch.

File: app/src/main/res/navigation/nav_graph.xml [hunk 3]
```diff
@@ -109,7 +111,9 @@
         app:destination="@id/quickSettingsSheetDialogFragment" />
     <action
         android:id="@+id/action_global_tabTrayDialogFragment"
-        app:destination="@id/tabTrayDialogFragment" />
+        app:destination="@id/tabTrayDialogFragment"
+        app:popUpTo="@id/tabTrayDialogFragment"
+        app:popUpToInclusive="true"/>
```
Comment (eliserichards) [inline app/src/main/res/navigation/nav_graph.xml:116]: Ditto here 😄

**Review by eliserichards — APPROVED** [review]
Fantastic! This is good to go

**Pull request merged**